c_secure_comm.c

```diff
@@ -477,28 +624,52 @@ session_key_list_t *send_session_key_req_via_TCP(SST_ctx_t *ctx) {
                 printf("Auth nonce verified!\n");
             }
             close(sock);
+            state = FINISHED;
             return session_key_list;
+        } else if (message_type == AUTH_ALERT) {
+            session_key_list->num_key = 0;
+            switch (data_buf[0]) {
+                case INVALID_DISTRIBUTION_KEY:
+                    printf("Error: Invalid Distribution Key\n");
+                    break;
+                case INVALID_SESSION_KEY_REQ:
+                    printf("Error: Invalid Session Key Request\n");
+                    break;
+                case UNKNOWN_INTERNAL_ERROR:
+                    printf("Error: Unknown Internal Error\n");
+                    break;
+                default:
+                    printf("Error: Unknown Code\n");
+                    break;
+            }
+            return NULL;
         }
     }
+    // Should not come here.
+    return NULL;
 }
 
-session_key_list_t *send_session_key_req_via_UDP(SST_ctx_t *ctx) {
-    session_key_list_t *s_key_list;
-    return s_key_list;
-    // TODO:(Dongha Kim) Implement this function.
-    error_exit("This function is not implemented yet.");
-}
+// session_key_list_t *send_session_key_req_via_UDP(SST_ctx_t *ctx) {
+//     session_key_list_t *s_key_list;
+//     return s_key_list;
+//     // TODO:(Dongha Kim) Implement this function.
```

Comment:
Move TODO to the beginning of this function.

Reply:
Also here. d2adfae8aa8ee4c09bd331c81d804798dc2f09f1